test/k8sT/Policies.go

```diff
@@ -827,78 +736,157 @@ EOF`, k, v)
+			res.ExpectSuccess("unable to apply manifest %s")
+
+			ciliumPod, err = kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+			Expect(err).Should(BeNil(), "cannot get CiliumPod")
+
+			err := kubectl.WaitforPods(namespace, "-l zgroup=testapp", 300)
+			Expect(err).To(BeNil(), "testapp pods are not ready after timeout")
+
+			err = kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
+			Expect(err).To(BeNil(), "testapp pods are not ready after timeout")
+
+			appPods = helpers.GetAppPods(apps, helpers.DefaultNamespace, kubectl, "id")
+			appPodsNS = helpers.GetAppPods(apps, namespace, kubectl, "id")
+
+			clusterIP, _, err = kubectl.GetServiceHostPort(helpers.DefaultNamespace, app1Service)
+			Expect(err).To(BeNil(), "Cannot get service on %q namespace", helpers.DefaultNamespace)
+
+			clusterIPNS, _, err = kubectl.GetServiceHostPort(namespace, app1Service)
+			Expect(err).To(BeNil(), "Cannot get service on %q namespace", namespace)
+
 		})
 
 		AfterEach(func() {
-			_ = kubectl.DeleteResource("netpol", fmt.Sprintf("%s -n%s", policy, namespace))
-			cleanupServerPodAndService(kubectl, podServer, service)
+			// Explicitly do not check results to avoid incomplete teardown of test.
+			_ = kubectl.Delete(policy)
+			_ = kubectl.Delete(l3Policy)
+			_ = kubectl.Delete(netpolNsSelector)
+
 		})
 
 		AfterAll(func() {
-			By("Deleting the namespace that was used for the pods")
-			err := kubectl.CoreV1().Namespaces().Delete(namespace, &metav1.DeleteOptions{})
-			Expect(err).NotTo(HaveOccurred(), "cannot delete %q namespace", namespace)
+			_ = kubectl.Delete(demoManifest)
+			_ = kubectl.NamespaceDelete(namespace)
 		})
 
-		It("should enforce policy based on NamespaceSelector", func() {
-			By("Testing pods can connect to both ports when no policy is present.")
-			testCanConnect(kubectl, namespace, "client-can-connect-80", service, 80, true)
-			testCanConnect(kubectl, namespace, "client-can-connect-81", service, 81, true)
-
-			By("Apply the network policy")
-			policyBeforeCreate := &networkingv1.NetworkPolicy{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: policy,
-				},
-				Spec: networkingv1.NetworkPolicySpec{
-					// Apply to server
-					PodSelector: metav1.LabelSelector{
-						MatchLabels: map[string]string{
-							"pod-name": podServer.Name,
-						},
-					},
-					// Allow traffic only from NS-B
-					Ingress: []networkingv1.NetworkPolicyIngressRule{{
-						From: []networkingv1.NetworkPolicyPeer{{
-							NamespaceSelector: &metav1.LabelSelector{
-								MatchLabels: map[string]string{
-									"ns-name": namespace,
-								},
-							},
-						}},
-					}},
-				},
+		It("Tests the same Policy in the different namespaces", func() {
+
+			By("Applying Policy in %q namespace", namespace)
+			_, err = kubectl.CiliumPolicyAction(
+				helpers.KubeSystemNamespace, policy, helpers.KubectlApply, 300)
+			Expect(err).Should(BeNil(), "L3 Policy cannot be applied in %q namespace", namespace)
```

Comment:
This error message duplicates the one on line 814 - differentiate the two for easier debugging.